### PR TITLE
Added clientDB connection

### PR DESCRIPTION
**Pull request by NJavokhir**
I added a clientDB connection to tunnel.go so that when calling tunnel all the client data comes as an array.

### CODE REVIEW

File: server/resource_server/repository/repository.go [hunk 1]
```diff
@@ -99,6 +100,15 @@ func (r *Repository) GetClientData(clientName string) (models.Client, error) {
...
 
+func (r *Repository) GetClientDataByBackendURL(backendURL string) ([]models.Client, error) {
+	fmt.Println("ABABABA", backendURL)
+    var clients []models.Client
+    if err := r.connection.Where("backend_uri = ?", backendURL).First(&clients).Error; err != nil {
+        return nil, err
+    }
+    return clients, nil
+}
+
```
Comment (ternakkode) [inline server/resource_server/repository/repository.go:111]: can we return just one value instead of array?

File: server/handlers/tunnel.go [hunk 2]
```diff
@@ -44,6 +45,11 @@ var (
 
 // Tunnel forwards the request to the service provider's backend
 func InitTunnel(w http.ResponseWriter, r *http.Request) {
+
+	r.Header.Set("BackendURL", "victus_backend.com")
```
Comment (ternakkode) [inline server/handlers/tunnel.go:49]: is it for testing purpose?

File: server/handlers/tunnel.go [hunk 3]
```diff
@@ -44,6 +45,11 @@ var (
...
+	clientPortalData := Ctl.GetClientDataByBackendURL(w, r)
```
Comment (ternakkode) [inline server/handlers/tunnel.go:50]: it should be using value from backend variable below, and i think you don't need to create a new controller, just call the service

File: server/resource_server/repository/repository.go (hunk 1 above)
Comment (NJavokhir) [reply server/resource_server/repository/repository.go:111]: Yes, we can

File: server/handlers/tunnel.go (hunk 2 above)
Comment (NJavokhir) [reply server/handlers/tunnel.go:49]: Yes, it is testing  value, we can send any backend url instead of it

File: server/handlers/tunnel.go (hunk 3 above)
Comment (NJavokhir) [reply server/handlers/tunnel.go:50]: Okay, I will

File: server/handlers/tunnel.go [hunk 4]
```diff
@@ -44,6 +45,10 @@ var (
 
 // Tunnel forwards the request to the service provider's backend
 func InitTunnel(w http.ResponseWriter, r *http.Request) {
+	newService := r.Context().Value("service").(interfaces.IService)
+	clients, err := newService.GetClientDataByBackendURL("victus_backend.com")
+	fmt.Println("client data", clients)
```
Comment (ternakkode) [inline server/handlers/tunnel.go:50]: remove this println

File: server/handlers/tunnel.go [hunk 5]
```diff
@@ -44,6 +45,10 @@ var (
 
 // Tunnel forwards the request to the service provider's backend
 func InitTunnel(w http.ResponseWriter, r *http.Request) {
+	newService := r.Context().Value("service").(interfaces.IService)
+	clients, err := newService.GetClientDataByBackendURL("victus_backend.com")
```
Comment (ternakkode) [inline server/handlers/tunnel.go:49]: can we try to call it after line 66 using backend variable?

**Pull request merged**